simavr/sim/run_avr.c

```diff
@@ -121,6 +121,63 @@ main(
 				display_usage(basename(argv[0]));
 		} else if (!strcmp(argv[pi], "-t") || !strcmp(argv[pi], "--trace")) {
 			trace++;
+		} else if (!strcmp(argv[pi], "--vcd-trace-name")) {
+			if(pi+1>=argc) {
+				fprintf(stderr, "%s: missing mandatory argument for %s.\n", argv[0], argv[pi]);
+				exit(1);
+			}
+			++pi;
+			strncpy(f.tracename, argv[pi], sizeof(f.tracename));
+		} else if (!strcmp(argv[pi], "--add-vcd-trace")) {
+			if(pi+1>=argc) {
+				fprintf(stderr, "%s: missing mandatory argument for %s.\n", argv[0], argv[pi]);
+				exit(1);
+			}
+			++pi;
+			struct {
+				char     kind[64];
+				uint8_t  mask;
+				uint16_t addr;
+				char     name[64];
+			} trace;
+			if(4 != sscanf(argv[pi], "%63[^=]=%63[^@]@0x%hx/0x%hhx", &trace.name[0], &trace.kind[0], &trace.addr, &trace.mask)) {
```

Comment:
Please use if<space>( -- also -- please don't use Yoda notation for conditions.

Reply:
Oh, and wrap super-long lines!

Reply:
Could you please clarify? By "Yoda notation" you probably mean to have `if(sscanf(...) != 4)`, but I don't understand the first part, "please use if( -- also --"?

Reply:
sorry github ate my less-than greather-than tag... I mean please use if+space+parenthesis -- in general also use spaces in expressions like if (pe + 1 >= argc) { ...